### PR TITLE
12-03 stream consistency

### PR DESCRIPTION
**Pull request by ailntps**

### CODE REVIEW

File: meeting_summary/2021-12-03-流计算端到端一致性概述.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+1. 早期的流计算平台无法保证一致性，如storm只保证at-least-once
+2. Lambda架构的流批一体处理平台的批处理部分用来对流处理的近似结果进行纠正。
+3. 一个流计算平台的一致性问题只要可以保证上游的数据重发和下游的算子去重即可
+
+参考资料：
+
+谷歌dataflow和beam实现[stream system](https://book.douban.com/subject/27080632/)
+
+谷歌流计算一致性[MillWheel](MillWheel: Fault-Tolerant Stream Processing atInternet Scale)
```
Comment (IcePigZDB) [inline meeting_summary/2021-12-03-流计算端到端一致性概述.md:26]: How about add link to pdf?

File: README.md [hunk 2]
```diff
@@ -5,4 +5,5 @@ DSCL weekly group meeting paper list.
 | :--------: | :------------------: | :----------------------------------------------------------: | :----------------------------------------------------------: |
 | 2021-10-29 | Database Transaction | Transaction, ACID, Distributed Transaction, 2PC, Consensus Algorithm |      [doc](./meeting_summary/2021-10-29-数据库事务.md)       |
 | 2021-11-05 |     LSM Tiering      |              LSM、Partition、Leveling、Tiering               | [doc](./meeting_summary/2021-11-05-基于LSM的KV存储写放大优化.md) |
+| 2021-12-03 |    stream system     |             stream,end to end consistency,Flink              |                                                              |
```
Comment (ehds) [inline README.md]: Please fill `Summary`  item. `[流计算端到端一致性概述](./meeting_summary/2021-12-03-流计算端到端一致性概述.md)`

**Review by ehds — CHANGES_REQUESTED** [review]

**Review by ehds — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -5,4 +5,6 @@ DSCL weekly group meeting paper list.
 | :--------: | :------------------: | :----------------------------------------------------------: | :----------------------------------------------------------: |
 | 2021-10-29 | Database Transaction | Transaction, ACID, Distributed Transaction, 2PC, Consensus Algorithm |      [doc](./meeting_summary/2021-10-29-数据库事务.md)       |
 | 2021-11-05 |     LSM Tiering      |              LSM、Partition、Leveling、Tiering               | [doc](./meeting_summary/2021-11-05-基于LSM的KV存储写放大优化.md) |
+| 2021-11-26 | determinacy database |                      Calvin，PWV，Aria                       |     [doc](./meeting_summary/2021-11-27-确定性数据库.md)      |
+| 2021-12-03 |    stream system     |             stream,end to end consistency,Flink              | [doc](./meeting_summary/2021-12-03-流计算端到端一致性概述.md) |
```
Comment (IcePigZDB) [inline README.md:9]:
```suggestion
| 2021-11-26 | Determinacy Database |                      Calvin，PWV，Aria                       |     [doc](./meeting_summary/2021-11-27-确定性数据库.md)      |
| 2021-12-03 |    Stream System     |             Stream, End to end consistency, Flink              | [doc](./meeting_summary/2021-12-03-流计算端到端一致性概述.md) |
```

Comment (ailntps) [reply README.md]: have fixed

**Comment by ailntps** [comment]
refix this

File: meeting_summary/2021-12-03-流计算端到端一致性概述.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+* 时间：2021.12.03
+* 分享人：王四
+* 关键字：Stream, End to end consistency, Flink
+* 分享PPT：[2021-12-03-流计算端到端一致性概述](./slides/2021-12-03-流计算端到端一致性概述.pdf)
+
+### 分享内容
+
+### 问题描述
+
+一个可以保证一致性（正确性）的流计算平台既可以处理流数据，也可以处理批数据。本次组会DataFlow、kafka、spark和Flink的一致性保证机制进行分析，总结流处理平台面对的一致性问题根本来源。
```
Comment (IcePigZDB) [inline meeting_summary/2021-12-03-流计算端到端一致性概述.md:12]:
```suggestion
一个可以保证一致性（正确性）的流计算平台既可以处理流数据，也可以处理批数据。本次组会对DataFlow、kafka、spark和Flink的一致性保证机制进行分析，总结流处理平台面对的一致性问题根本来源。
```

File: meeting_summary/2021-12-03-流计算端到端一致性概述.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+### 分享内容
+
+### 一、背景：
+
+1. 早期的流计算平台无法保证一致性，如storm只保证at-least-once
+2. Lambda架构的流批一体处理平台的批处理部分用来对流处理的近似结果进行纠正。
```
Comment (IcePigZDB) [inline meeting_summary/2021-12-03-流计算端到端一致性概述.md:19]:
```suggestion
2. Lambda架构的流批一体处理平台的批处理部分用来对流处理的近似结果进行纠正
```

File: meeting_summary/2021-12-03-流计算端到端一致性概述.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+3. 一个流计算平台的一致性问题只要可以保证上游的数据重发和下游的算子去重即可
```
Comment (IcePigZDB) [inline meeting_summary/2021-12-03-流计算端到端一致性概述.md:20]:
```suggestion
1. 早期的流计算平台无法保证一致性，如storm只保证at-least-once。
2. Lambda架构的流批一体处理平台的批处理部分用来对流处理的近似结果进行纠正。
3. 一个流计算平台的一致性问题只要可以保证上游的数据重发和下游的算子去重即可。
```

**Review by IcePigZDB — APPROVED** [review]

**Pull request merged**